docs/proposals/metadata/external-host-tags.md

```diff
@@ -0,0 +1,76 @@
+# External Host Tags collection
+
+- Authors: Massimiliano Pippi
+- Date: 2018-01-15
+- Status: review
+- [Discussion](https://github.com/DataDog/datadog-agent/pull/1053)
+
+## Overview
+
+Some checks (at this moment there are only 2 of them: [vsphere][] and [openstack][])
+expose a method with the signature `get_external_host_tags()` that's called by
+the Agent whenever it's time to build and send the metadata payload. The way the
+new Agent collects metadata has drastically changed and this feature wasn't been
+ported yet, thus such checks are currently not working.
+
+## Problem
+
+Previous versions of the Agent took advantage of the dynamic nature of Python
+and invoked the collection method [like this][agent5-collection], an alternative
+must be defined to provide the new Agent with the same feature.
+
+## Constraints
+
+Changes to the checks must be compatible with both the agents 5.x and 6.x.
+
+## Recommended Solution
+
+### Change the collection model from "Pull" to "Push"
+
+Instead of invoking a method exposed by the checks at every metadata collection,
+let the checks push data to the Agent. External tags would be stored and used by
+the Agent when it's time to send the metadata payload.
+
+This would only require to add a new method to the Python Checks API exposed by
+the Agent. On the check side, `get_external_host_tags` would be preserved for
+backward compatibility and some logic would be added to send host tags during
+metrics collection (some strategy might be needed to avoid sending external host
+tags at every collection cycle, this might be different from one check to another).
```

Comment:
If I understand correctly you said so in the first paragraph of this section, but just to repeat it: it's important for the Agent to control the rate at which it actually sends external host tags to DD. (as these payloads are more costly to ingest than metrics payloads for example)

Reply:
I'm definitely eye-balling this but it doesn't seem too elaborate to store timestamped external host metadata in a map (possibly mapped by check or similar), and only adding the additional metadata to the payload every nth iteration or n secs.